source/MRViewer/MRObjectTransformWidget.cpp

```diff
@@ -686,7 +764,8 @@ void ObjectTransformWidget::addXf_( const AffineXf3f& addXf )
     approvedChange_ = true;
     if ( addXfCallback_ )
         addXfCallback_( addXf );
-    controlsRoot_->setXf( addXf * controlsRoot_->xf() );
+    if ( findCurrentObjIndex_() >= 3 || axisTransformMode_ == Translation )
+        controlsRoot_->setXf( addXf * controlsRoot_->xf() );
```

Comment:
why this if? we are already in `processTranslation_` function

Reply:
This function can also be called from `processScaling_` and `processRotation_`. There's no single parameter that indicates scale editing.